### ISSUE TITLE
Resource Groups: Create TabletServer property like sserver.group that can be used for assignment / balancing

### ISSUE THREAD

**Issue by dlmarion** [apache/accumulo#3459]

**Comment by dlmarion** [comment]
Users will use these group properties to define resource groups, sets of compactors, sservers, and tservers, upon which their application(s) will use to interact with their tables.

For compactions, the compaction service property defines the configuration of compaction size to compactor group. This will enable a user to specify X compactors for one size and Y compactors for another. A table property is used to determine which service configuration is used for the tables compactions.

For tablet assignment the new `tserver.group` property will be used to determine which set of tservers will be passed to the table balancer.

For clients performing eventual scans, the client scan server selector configuration determines which group of ScanServers will be used to execute the scans. 

System tables will always be hosted, therefore at least 1 tablet server is required. Compactions will be performed using Compactors, therefore at least 1 Compactor is required.

I'm intending on doing to the following for this solution:

1. Create `tserver.group` property
2. Rename `compactor.queue` property to `compactor.group` for consistency. 
3. Create a pre-defined `system` group for defining tablet servers and compactors for system tables.
4. Use the pre-defined `default` group for user tables.
5. Modify cluster.yaml and supporting code to enforce `system` group members. The new file structure will look something like:
```
manager:
  - localhost

monitor:
  - localhost1

gc:
  - localhost

tserver:
  system:
    - localhost1
  default:
    - localhost2

compactor:
  system:
    - localhost1
  default:
    - localhost2
```

**Comment by dlmarion** [comment]
Closed by #3591
